examples/hackweek_demo/plot_pnl.py

```diff
@@ -112,93 +113,134 @@ def get_wallet_from_onchain_trade_info(
+            weighted_average_share_price.loc[row, :] = weighted_average_share_price.loc[row - 1, :].where(
+                ~cols, new_avg, axis=0
             )
 
+        agent_info[agent] = Info(
+            wallet=wallet,
+            positions=positions,
+            deltas=deltas,
+            open_share_price=share_price_on_increases,
+        )
+    return agent_info
+
+
+def calculate_pnl(pool_config, pool_info, checkpoint_info):
+    """Calculate pnl for all agents."""
+    #  pylint: disable=too-many-locals
+    session = postgres.initialize_session()
+    all_agent_info = get_agent_info(session)
+    position_duration = pool_config.positionDuration.iloc[0]
+
+    for _, agent_info in all_agent_info.items():
+        wallet, positions, deltas, open_share_price = agent_info
+        for block in agent_info.wallet.index:
+            position = positions.loc[block]
+            current_wallet = wallet.loc[block, :]
+            state = pool_info.loc[block]
+            if block in checkpoint_info.index:  # a checkpoint exists
+                current_checkpoint = checkpoint_info.loc[block]
+                maturity = current_checkpoint["timestamp"] + pd.Timedelta(seconds=position_duration)
+            else:
+                maturity = None
+
             spot_price = calculate_spot_price(
-                row.share_reserves,
-                row.bond_reserves,
-                row.lp_total_supply,
-                row.maturity_time,
-                row.block_timestamp,
+                state.shareReserves,
+                state.bondReserves,
+                state.lpTotalSupply,
+                maturity,
+                current_wallet.timestamp,
                 position_duration,
             )
-            # print(f"{spot_price=}")
 
-            # LP value (TODO: check why certain agents own more than 100% of the pool)
-            total_lp_value = row.share_reserves / 1e18 * row.share_price / 1e18 + row.bond_reserves / 1e18 * spot_price
-            share_of_pool = float(agent_wallets[agent].lp_tokens) / (row.lp_total_supply / 1e18)
-            # print(f"{float(agent_wallets[agent].lp_tokens)=}")
-            # print(f"{row.lp_total_supply/1e18=}")
-            pnl = share_of_pool * total_lp_value
-
-            # for each LONG
-            for _, long in agent_wallets[agent].longs.items():
-                pnl += float(long.balance) * spot_price
-
-            # for each SHORT
-            for _, short in agent_wallets[agent].shorts.items():
-                pnl += float(short.balance) * (1 - spot_price)
-
-            new_pnl[f"{agent_shortname}"] = pnl
-        pnl_data.append(new_pnl)
-        time_data.append(row["timestamp"])
+            pnl = 0
+            for position_name in positions.columns:
+                if position_name.startswith("LP"):
+                    # LP value (TODO: check why certain agents own more than 100% of the pool)
```

Comment:
whhaaat

Reply:
they used to, the tripwire isn't getting triggerred anymore

Reply:
phew